server/app/services/program/predicate/LeafAddressServiceAreaExpressionNode.java

```diff
@@ -0,0 +1,78 @@
+package services.program.predicate;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+import services.question.types.QuestionDefinition;
+
+/** Represents an assertion that an address question is within a given service area. */
+@JsonTypeName("leafAddressServiceArea")
+@AutoValue
+public abstract class LeafAddressServiceAreaExpressionNode
+    implements ConcretePredicateExpressionNode {
+
+  @JsonCreator
+  public static LeafAddressServiceAreaExpressionNode create(
+      @JsonProperty("questionId") long questionId,
+      @JsonProperty("serviceAreaId") String serviceAreaId) {
+    return builder().setQuestionId(questionId).setServiceAreaId(serviceAreaId).build();
+  }
+
+  /** The ID of the address {@link services.question.types.QuestionDefinition} this node checks. */
+  @JsonProperty("questionId")
+  public abstract long questionId();
+
+  /** The string label of the service area the address should be checked for inclusion in. */
```

Comment:
suggestion (non-blocking): cleanup references to label in this file.

Reply:
oh woops, thanks!